### PR TITLE
DR-1586 Make sure that we properly handle the case where a user doesn't have permissions to view policies

### PR DESCRIPTION
**Pull request by nmalfroy**
...e.g. view who belongs to a specific policy.

Note: this is for the details side bar on the snapshot view

This also:
- Makes sure that we can add/remove stewards
- Refactors the code to make it easier and consistent to add/remove snapshot roles
- Does a drive by fix which resets the UI state when you log out (so that when you log in as someone else, it doesn't show you your previous snapshots and datasets while it loads)

### CODE REVIEW

File: src/components/DetailViewHeader.jsx [hunk 1]
```diff
@@ -29,26 +29,28 @@ const styles = (theme) => ({
 
 export class DetailViewHeader extends React.PureComponent {
   static propTypes = {
-    addCustodian: PropTypes.func,
+    addSteward: PropTypes.func,
     addReader: PropTypes.func,
```
Comment (myessail) [inline src/components/DetailViewHeader.jsx:32]: if you wanna fix these linting warnings since you're touching this file, i would appreciate 🙂

**Review by myessail — APPROVED** [review]
💪

File: src/components/DetailViewHeader.jsx (hunk 1 above)
Comment (nmalfroy) [reply src/components/DetailViewHeader.jsx:32]: ah touché.  Will do

**Review by fboulnois — APPROVED** [review]
Seems pretty clear, lgtm

**Pull request merged**